app/forms/key_people_form.rb

```diff
@@ -9,10 +10,52 @@ def initialize(transient_registration)
 
   def submit(params)
     # Assign the params for validation and pass them to the BaseForm method for updating
-    # TODO: Define allowed params, eg self.field = params[:field]
-    # TODO: Include attributes to update in the attributes hash, eg { field: field }
-    attributes = {}
+    self.first_name = params[:first_name]
+    self.last_name = params[:last_name]
+    self.dob_day = params[:dob_day].to_i
+    self.dob_month = params[:dob_month].to_i
+    self.dob_year = params[:dob_year].to_i
+
+    self.key_person = add_key_person
+    self.date_of_birth = key_person.date_of_birth
+
+    attributes = { keyPeople: [key_person] }
 
     super(attributes, params[:reg_identifier])
   end
+
+  validates :first_name, presence: true, length: { maximum: 35 }
+  validates :last_name, presence: true, length: { maximum: 35 }
+  validate :old_enough?
+  validates_with DateOfBirthValidator
+
+  private
+
+  def add_key_person
+    KeyPerson.new(first_name: first_name,
+                  last_name: last_name,
+                  dob_day: dob_day,
+                  dob_month: dob_month,
+                  dob_year: dob_year,
+                  person_type: "key")
+  end
+
+  def old_enough?
```

Comment:
It's my belief that methods phrased as questions should only return `true`, `false` or throw an exception.

So I would decide what result you want to return in the case of `date_of_birth` not being present, and ensure false is returned if `date_of_birth < age_cutoff_date`

Reply:
This should be fixed now.